### PR TITLE
Patch for molden (ambor) when using gcc>10

### PR DESCRIPTION
**Pull request by hpcnpatel**
While facing compilations errors due to features introduced in GCC 10&11, a suggestion was to add the -fallow-argument-mismatch compilation flag to gfortran. This action needs to be performed in the ambfor makefile.

### CODE REVIEW

File: var/spack/repos/builtin/packages/molden/package.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ class Molden(MakefilePackage):
     build_targets = ['clean', 'all']
 
     patch('for_aarch64.patch', when='target=aarch64:')
+    patch('ambor_gcc10.patch', when='%gcc@10:')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/molden/package.py]: How about doing something like to get rid of the patch?
```python
    def flag_handler(self, name, flags):
        if name == 'fflags':
            if self.spec.satisfies('%gcc@10:'):
                if flags is None:
                    flags = []
                flags.append('-fallow-argument-mismatch')

        return (flags, None, None)
```

Comment (hpcnpatel) [reply var/spack/repos/builtin/packages/molden/package.py]: ya, makes sense. I will update the file. Thanks

**Comment by michaelkuhn** [comment]
Sorry I missed this before but the `package.py` already contains `-fallow-argument-mismatch`, see: https://github.com/spack/spack/blob/07762ea274b72455cc5165bfa1be13aa8d77da5e/var/spack/repos/builtin/packages/molden/package.py#L55-L56

It fails for me, though, since a space is missing:
```
gfortran: error: unrecognized command-line option '-funroll-loops-fallow-argument-mismatch'; did you mean '-fno-allow-argument-mismatch'?
```

**Comment by hpcnpatel** [comment]
> Sorry I missed this before but the `package.py` already contains `-fallow-argument-mismatch`, see:
> 
> https://github.com/spack/spack/blob/07762ea274b72455cc5165bfa1be13aa8d77da5e/var/spack/repos/builtin/packages/molden/package.py#L55-L56
> 
> It fails for me, though, since a space is missing:
> 
> ```
> gfortran: error: unrecognized command-line option '-funroll-loops-fallow-argument-mismatch'; did you mean '-fno-allow-argument-mismatch'?
> ```

Even with the space, what I had seen is, the flag was not correctly passed to the ambor. You could give it try.

**Comment by michaelkuhn** [comment]
You're right, even with a space it doesn't work. I'm just a bit surprised that the build does not fail for you due to the typo.

Would you please remove the duplicate line(s) and only keep the flag handler? Then we can merge this.

**Comment by hpcnpatel** [comment]
> You're right, even with a space it doesn't work. I'm just a bit surprised that the build does not fail for you due to the typo.
> 
> Would you please remove the duplicate line(s) and only keep the flag handler? Then we can merge this.

done, thanks for your time

**Review by michaelkuhn — DISMISSED** [review]
LGTM
@dev-zero any comments?

**Review by dev-zero — DISMISSED** [review]
LGTM, thanks!

**Comment by dev-zero** [comment]
If you can fix the style issues, the bot should be able to merge automatically.

**Comment by michaelkuhn** [comment]
Sorry, totally missed those. Seems to be an easy fix:
```
var/spack/repos/builtin/packages/molden/package.py:69: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/molden/package.py:70: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/molden/package.py:71: [E303] too many blank lines (2)
```

**Comment by michaelkuhn** [comment]
Still fails:
```
var/spack/repos/builtin/packages/molden/package.py:69: [E301] expected 1 blank line, found 0
```
(The empty line after the `flag_handler` function is missing.)

**Comment by hpcnpatel** [comment]
> Still fails:
> 
> ```
> var/spack/repos/builtin/packages/molden/package.py:69: [E301] expected 1 blank line, found 0
> ```
> 
> (The empty line after the `flag_handler` function is missing.)

this is rather frustrating. 

> Still fails:
> 
> ```
> var/spack/repos/builtin/packages/molden/package.py:69: [E301] expected 1 blank line, found 0
> ```
> 
> (The empty line after the `flag_handler` function is missing.)

This is rather frustrating, dealing with blank lines. Pheww

**Comment by haampie** [comment]
Yeah, it should be fixed by tools not by humans, hopefully we get that 🔜 ™️, `spack style --fix` already does this partiallly

**Review by michaelkuhn — DISMISSED** [review]

**Comment by michaelkuhn** [comment]
Looks like another error snuck in (you can also check this locally using `spack style`):
```
var/spack/repos/builtin/packages/molden/package.py:69: [W293] blank line contains whitespace
```
Could you please make this hopefully last change? Alternatively, I can push a fix to your PR branch.

**Review by michaelkuhn — APPROVED** [review]

**Pull request merged**